### ISSUE TITLE
Camera causes app crash when entered from notifications

### ISSUE THREAD

**Issue by Robrowski** [Robrowski/GeoSnap#8]
# Circumstances:
1. open app
2. Hit the menu option to "Test Discovered Snaps Notification"
3. Without closing the app (specifically don't leave MainActivity), click on the Notification

# What I know is happening:
1. onPause is called in old fragment (currently owns camera)
2. New activity starts up as per the intent in the notification
3. New fragment attempts to start and claim camera
4. CRASH
5. Hypothetically, if it didn't crash, old fragment onDestroy gets called at the very end

# Why I haven't fixed it:

The solution is SUPPOSED to be releasing the camera in the "onPause" function (according to API), but the problem is onPause gets called other times too. Releasing effectively kills the mCamera object. Therefore, the camera gets RELEASED and gone, even though the surface view will be relying on that later. EX. turning your screen on then off freezes the camera. 

If we can somehow restart/reinstantiate the SurfaceView used for the camera preview with the new camera object FROM the onResume of the CameraPreviewFragment, all issues solved.

**Comment by Robrowski** [comment]
- found the solution, just need time to put it in

https://github.com/pikanji/CameraPreviewSample/tree/master/src/net/pikanji/camerapreviewsample

**Comment by mscosti** [comment]
This is the same repository I used when for switching to dynamic image preview sizing, It's possible I didn't get everything quite right.  

If this does not work, I bet we could go back to the version before dynamic previewing, since I started to notice a similar bug after implementing it, but it wasn't as consistent as this one.  I would sometimes crash when after clicking a snap to go to snapview and then crash when going back to the list view.
